newIDE/app/src/MainFrame/index.js

```diff
@@ -3867,17 +3880,47 @@ const MainFrame = (props: Props) => {
           onLaunchPreview={
             hotReloadPreviewButtonProps.launchProjectDataOnlyPreview
           }
-          onClose={options => {
+          onClose={async options => {
+            if (hasUnsavedChanges) {
+              const response = await showConfirmation({
+                title: t`Leave the customization?`,
+                message: t`Do you want to quit the customization? All your changes will be lost.`,
+                confirmButtonLabel: t`Leave`,
+              });
+
+              if (!response) {
+                return;
+              }
+            }
+
             setQuickCustomizationDialogOpenedFromGameId(null);
-            if (options && options.tryAnotherGame) {
-              // Close the project so the user is back at where they can chose a game to customize
-              // which is probably the home page.
-              closeProject();
-              openHomePage();
+            closeProject();
+            openHomePage();
+            if (!hasUnsavedChanges) {
+              navigateToRoute('build');
             }
           }}
           onlineWebExporter={quickPublishOnlineWebExporter}
-          onSaveProject={saveProject}
+          onSaveProject={async () => {
+            // Automatically save project to the cloud if no provider is set.
+            const storageProvider = getStorageProvider();
+            console.log('storageProvider', storageProvider);
+            if (
+              storageProvider.internalName === 'Empty' ||
+              storageProvider.internalName === 'UrlStorageProvider'
+            ) {
+              getStorageProviderOperations(CloudStorageProvider);
+              saveProjectAsWithStorageProvider({
+                requestedStorageProvider: CloudStorageProvider,
+                forcedSavedAsLocation: {
```

Comment:
I went for a new prop here to force the name of the project and skip the "Select the Cloud Project name" dialog. There is no unicity on the name, so I think we're fine

Reply:
Ok 👍